src/formatDate.js

```diff
@@ -9,6 +9,64 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldDateSeparator = fromFormat[3];
+  const splitFromFormat = date.split(oldDateSeparator);
+  const newFormatDate = [];
+  const year = [];
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year.push(splitFromFormat[i]);
+        break;
+
+      case 'M':
+        month = splitFromFormat[i];
+        break;
+
+      case 'D':
+        day = splitFromFormat[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i][0]) {
+      case 'Y':
+        const yearNewFormatLength = toFormat[i].length;
+
+        if (year[0].length === yearNewFormatLength) {
+          newFormatDate.push(...year);
+        } else if (year[0].length > yearNewFormatLength) {
+          newFormatDate.push(year[0].slice(2));
+        } else if (year[0].length < yearNewFormatLength && year[0] < 30) {
+          year.unshift(20);
+          newFormatDate.push(year.join(''));
+        } else if (+year[0] === 30 || year[0] > 30) {
+          year.unshift(19);
+          newFormatDate.push(year.join(''));
+        }
+        break;
+
+      case 'M':
+        newFormatDate.push(month);
+        break;
+
+      case 'D':
+        newFormatDate.push(day);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return newFormatDate.join(toFormat[3]);
```

Comment:
The resulting date is joined using the new format separator, but there is no check to ensure that the `toFormat` array has a separator at index 3. You should validate that `toFormat` includes a separator before using it.